test/unit/MemberRoles/stateManagement.js

```diff
@@ -0,0 +1,109 @@
+    const { memberRoles } = this.contracts;
+
+    const authorizedAddress = await memberRoles.authorized(Role.Member);
+    expect(authorizedAddress).to.be.equal(AddressZero);
+  });
+
+  it('should return length of all roles', async function () {
+    const { memberRoles } = this.contracts;
+
+    const [unAssignedMembersLength, advisoryBoardMembersLength, membersMembersLength] =
+      await memberRoles.getMemberLengthForAllRoles();
+    expect(unAssignedMembersLength).to.be.equal(0);
+    expect(advisoryBoardMembersLength).to.be.equal(1);
+    expect(membersMembersLength).to.be.equal(membersCount);
+  });
+
+  it('should return members length', async function () {
+    const { memberRoles } = this.contracts;
+
+    const membersLength = await memberRoles.membersLength(Role.Member);
+    expect(membersLength).to.be.equal(membersCount);
+  });
+
+  it('should return member at index', async function () {
+    const { memberRoles } = this.contracts;
+    const {
+      members: [member],
+    } = this.accounts;
+
+    const [memberAddress, isActive] = await memberRoles.memberAtIndex(Role.Member, 0);
+    expect(member.address).to.be.equal(memberAddress);
+    expect(isActive).to.be.equal(true);
+  });
+
+  it('should clear storage', async function () {
+    const { memberRoles } = this.contracts;
+
+    await expect(memberRoles.storageCleanup()).to.not.reverted;
+  });
+
+  it('should check the role of a member', async function () {
+    const { memberRoles } = this.contracts;
+    const {
+      members: [member],
+      nonMembers: [nonMember],
+      advisoryBoardMembers: [advisoryBoardMember],
+    } = this.accounts;
+
+    const isMemberMember = await memberRoles.checkRole(member.address, Role.Member);
+    const isMemberNonMember = await memberRoles.checkRole(member.address, Role.NonMember);
+    const isMemberABMember = await memberRoles.checkRole(member.address, Role.AdvisoryBoard);
+
+    expect(isMemberMember).to.be.equal(true);
+    expect(isMemberNonMember).to.be.equal(true);
+    expect(isMemberABMember).to.be.equal(false);
+
+    const isNonMemberMember = await memberRoles.checkRole(nonMember.address, Role.Member);
+    const isNonMemberNonMember = await memberRoles.checkRole(nonMember.address, Role.NonMember);
+    const isNonMemberABMember = await memberRoles.checkRole(nonMember.address, Role.AdvisoryBoard);
+
+    expect(isNonMemberMember).to.be.equal(false);
+    expect(isNonMemberNonMember).to.be.equal(true);
+    expect(isNonMemberABMember).to.be.equal(false);
+
+    const isABMemberMember = await memberRoles.checkRole(advisoryBoardMember.address, Role.Member);
+    const isABMemberNonMember = await memberRoles.checkRole(advisoryBoardMember.address, Role.NonMember);
+    const isABMemberABMember = await memberRoles.checkRole(advisoryBoardMember.address, Role.AdvisoryBoard);
+
+    expect(isABMemberMember).to.be.equal(true);
+    expect(isABMemberNonMember).to.be.equal(true);
```

Comment:
also odd that it's true

Reply:
may be "correct" but warrants at least a comment to clarify why since it looks counterintuitive

Reply:
@MilGard91